sdk/nodejs/resource.ts

```diff
@@ -331,7 +331,8 @@ export class Output<T> {
      *
      * Only callable on the outside.
      */
-    /* @internal */ public readonly resources: () => Set<Resource>;
+    // tslint:disable-next-line:variable-name
+    /* @internal */ public readonly __resources: Promise<Set<Resource>>;
```

Comment:
I changed the name of this property after discussion with @CyrusNajmabadi. The change ought to make it easier to catch upstream code that was peeking at `resources()`.

Reply:
Are we sure this is versioning-safe?  I feel like we hit assingability issues across versions when we make changes like this.

Reply:
This whole change may not be versioning-safe. We'll need to think through the implications here.

Reply:
Okay, I've adjusted things s.t. we should be backwards-compatible here.

Reply:
(in nodejs)